game-core/src/main/java/games/strategy/triplea/delegate/battle/steps/change/RemoveUnprotectedUnits.java

```diff
@@ -0,0 +1,173 @@
+package games.strategy.triplea.delegate.battle.steps.change;
+
+import static games.strategy.triplea.delegate.battle.BattleStepStrings.REMOVE_UNESCORTED_TRANSPORTS;
+
+import games.strategy.engine.data.Change;
+import games.strategy.engine.data.GamePlayer;
+import games.strategy.engine.data.Unit;
+import games.strategy.engine.data.changefactory.ChangeFactory;
+import games.strategy.engine.delegate.IDelegateBridge;
+import games.strategy.triplea.Properties;
+import games.strategy.triplea.delegate.ExecutionStack;
+import games.strategy.triplea.delegate.Matches;
+import games.strategy.triplea.delegate.battle.BattleActions;
+import games.strategy.triplea.delegate.battle.BattleState;
+import games.strategy.triplea.delegate.battle.steps.BattleStep;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.EnumSet;
+import java.util.List;
+import java.util.function.Predicate;
+import lombok.AllArgsConstructor;
+import org.triplea.java.collections.CollectionUtils;
+
+@AllArgsConstructor
+public class RemoveUnprotectedUnits implements BattleStep {
+
+  private static final long serialVersionUID = 4357860848979564096L;
+
+  protected final BattleState battleState;
```

Comment:
Do these class variables have to be 'protected', can they be private?

Having child classes modify parent state is a bad form of coupling. It makes program flow very hard to reason about as you have to know the program state to know what will execute and how. Having class variables be uniformly private helps with this and avoiding inheritance.

Reply:
They can be private.  Leaving them as protected just makes accessing them easier in the child classes.  This will just change the code from `this.battleState.getUnits()` to `this.getBattleState().getUnits()` so I'm not sure how useful it is to make them private.

Reply:
Oops, ignore my comment.  I was thinking that this class had children classes but it doesn't.  I'll change them to private.